control_panel_api/authentication.py

```diff
@@ -39,9 +39,14 @@ def authenticate(self, request):
             raise AuthenticationFailed('JWT missing "sub" field')
 
         try:
-            user = User.objects.get(auth0_id=sub)
+            user = User.objects.get(pk=sub)
 
         except User.DoesNotExist:
-            raise AuthenticationFailed(f'No such user: {sub}')
+            user = User.objects.create(
+                pk=sub,
+                username=decoded.get(settings.OIDC_FIELD_USERNAME),
+                email=decoded.get(settings.OIDC_FIELD_EMAIL),
+                name=decoded.get(settings.OIDC_FIELD_NAME),
```

Comment:
Are these fields ok to set to `None` if the field is not in the JWT?

Reply:
name and email are `blank=True` and `username` is required and must be unique.